internal/app/go-filecoin/internal/submodule/storage_miner_node.go

```diff
@@ -0,0 +1,508 @@
+		types.NewGasPrice(1),
+		types.NewGasUnits(300),
+		true,
+		storagemarket.PreCommitSector,
+		precommitParams,
+	)
+	if err != nil {
+		return cid.Undef, err
+	}
+
+	err = <-cerr
+	if err != nil {
+		return cid.Undef, err
+	}
+
+	return mcid, nil
+}
+
+// WaitForPreCommitSector blocks until the pre-commit sector message is mined
+// into a block, returning the block's height and message's exit code (or an
+// error if one is encountered).
+func (m *StorageMinerNodeAdapter) WaitForPreCommitSector(ctx context.Context, mcid cid.Cid) (uint64, uint8, error) {
+	return m.waitForMessageHeight(ctx, mcid)
+}
+
+// SendProveCommitSector creates a commit sector message and places it into the
+// nodes message pool and returns its identity (or an error, if one is
+// encountered).
+func (m *StorageMinerNodeAdapter) SendProveCommitSector(ctx context.Context, sectorID uint64, proof []byte, deals ...uint64) (cid.Cid, error) {
+	dealIds := make([]types.Uint64, len(deals))
+	for i, deal := range deals {
+		dealIds[i] = types.Uint64(deal)
+	}
+
+	info := &types.SectorProveCommitInfo{
+		Proof:    proof,
+		SectorID: types.Uint64(sectorID),
+		DealIDs:  dealIds,
+	}
+
+	commitParams, err := abi.ToEncodedValues(info)
+	if err != nil {
+		return cid.Undef, err
+	}
+
+	mcid, cerr, err := m.messaging.Outbox.Send(
+		ctx,
+		m.workerAddr,
+		m.minerAddr,
+		types.ZeroAttoFIL,
+		types.NewGasPrice(1),
+		types.NewGasUnits(300),
+		true,
+		storagemarket.CommitSector,
+		commitParams,
+	)
+	if err != nil {
+		return cid.Undef, err
+	}
+
+	err = <-cerr
+	if err != nil {
+		return cid.Undef, err
+	}
+
+	return mcid, nil
+}
+
+// WaitForProveCommitSector blocks until the provided pre-commit message has
+// been mined into the chain, producing the height of the block in which the
+// message was mined (and the message's exit code) or an error if any is
+// encountered.
+func (m *StorageMinerNodeAdapter) WaitForProveCommitSector(ctx context.Context, mcid cid.Cid) (uint64, uint8, error) {
+	return m.waitForMessageHeight(ctx, mcid)
+}
+
+// GetSealTicket produces the seal ticket used when pre-committing a sector at
+// the moment it is called
+func (m *StorageMinerNodeAdapter) GetSealTicket(ctx context.Context) (storage.SealTicket, error) {
+	ts, err := m.chain.ChainReader.GetTipSet(m.chain.ChainReader.GetHead())
+	if err != nil {
+		return storage.SealTicket{}, xerrors.Errorf("getting head ts for SealTicket failed: %w", err)
+	}
+
+	h, err := ts.Height()
+	if err != nil {
+		return storage.SealTicket{}, err
+	}
+
+	r, err := m.chain.State.SampleRandomness(ctx, types.NewBlockHeight(h-consensus.FinalityEpochs))
+	if err != nil {
+		return storage.SealTicket{}, xerrors.Errorf("getting randomness for SealTicket failed: %w", err)
+	}
+
+	return storage.SealTicket{
+		BlockHeight: h,
+		TicketBytes: r,
+	}, nil
+}
+
+// GetSealSeed is used to acquire the seal seed for the provided pre-commit
+// message, and provides channels to accommodate chain re-orgs. The caller is
```

Comment:
> It seems like this should be hardcoded within this function rather than set as a parameter. Interested to hear your reasoning here.

@acruikshank - would you field this question plz?

Reply:
The goal here (and it's been a source of controversy between @laser and me) is to let go-storage-miner own as much of interactive PoRep as is reasonably possible. In this case it means that go-storage-miner owns the PoRep delay constant instead of requiring that implementations agree upon it.